### ISSUE TITLE
Truncated strings in `MultiSelect`

### ISSUE THREAD

**Issue by SimonFinney** [carbon-design-system/carbon#2966]
## Truncated strings in `MultiSelect`

Our globalisation testing team is raising defects related to tooltips for truncated strings, for example the `MultiSelect` component -

<img width="363" alt="MultiSelect Label" src="https://user-images.githubusercontent.com/3846874/59102349-b4b2fb00-8923-11e9-98f3-3f8812b8c38a.png">

In my opinion, I don't think there should be tooltips for scenarios like these, but rather that the component's width is adjusted based on the content.

I'm curious as to the team's thoughts around this, and whether we can collaborate on an ideal solution.

**Comment by emyarod** [comment]
by default the width of the container determines the width of the component. the label is truncated in cases where the content width is greater than the container width

https://codesandbox.io/s/codesandbox-xl2yn

**Comment by dakahn** [comment]
@aagonzales Is our guidance that the input's width should be made to fit the label, or that the labels length should be limited to accommodate the input's width?

@SimonFinney totally agree that a tooltip isn't what we want here ♿️

**Comment by aagonzales** [comment]
By default the component width should change depending on content however, there are use cases where the width of the dropdown could be fixed or maxed out in which case a truncation pattern would be needed. And agreed tooltip is not the ideal scenario.

Such scenarios include:
- constrained by layout such as inside another component (tile, form etc).
- screen size (like mobile)
- user generated content that is excessively long

Potential solutions: 
- make the dropdown height larger and respond to content.
- the content is truncated and then the option when appearing inside the menu itself is not.
- Another option is to `...` then make the content horizontally scrollable inside the dropdown field area.

**Comment by SimonFinney** [comment]
Just to revive this - what's the outcome? Will the `Dropdown` be enhanced to handle this issue or do we need to make adjustments on our side?

**Comment by grahamharper** [comment]
Here in 2024 to revive this as a tooltip via `title` attribute is not an ideal solution (as @aagonzales [said](https://github.com/carbon-design-system/carbon/issues/2966#issuecomment-501046647)) as this is not an accessible solution for sighted, keyboard-only users.

Truncation should not be the strategy here, instead the solution is two-fold:
1. When open, allow the listbox to be wider than the input, as wide as it needs to be (a limit could be applied to avoid excessively long line length e.g. `60ch` or 100% of the viewport whichever is smaller, because the viewport could be narrower and the listbox should not extend beyond the bounds of the viewport)
2. Once the width limit is reached, the item label text should wrap within its row (this is how the native `select` control handles long labels on iOS)

@SimonFinney what did you end up doing?

**Comment by grahamharper** [comment]
A prototype illustrating what I described above:


https://github.com/carbon-design-system/carbon/assets/2853295/cce26fe1-59eb-418a-b86d-4a4aed2d0d30
